backend/dataall/modules/s3_datasets/services/dataset_location_service.py

```diff
@@ -135,3 +134,42 @@ def _delete_dataset_folder_read_permission(session, dataset: S3Dataset, location
         }
         for group in permission_group:
             ResourcePolicyService.delete_resource_policy(session=session, group=group, resource_uri=location_uri)
+
+    @staticmethod
+    def get_folder_restricted_information(folder: DatasetStorageLocation):
+        try:
+            context = get_context()
+            with context.db_engine.scoped_session() as session:
+                ResourcePolicyService.check_user_resource_permission(
+                    session=session,
+                    username=context.username,
+                    groups=context.groups,
+                    resource_uri=folder.locationUri,
+                    permission_name=GET_DATASET_FOLDER,
+                )
+                dataset = DatasetRepository.get_dataset_by_uri(session, folder.datasetUri)
+                return {
+                    'AwsAccountId': dataset.AwsAccountId,
+                    'region': dataset.region,
+                    'S3BucketName': dataset.S3BucketName,
+                    'GlueDatabaseName': dataset.GlueDatabaseName,
+                    'IAMDatasetAdminRoleArn': dataset.IAMDatasetAdminRoleArn,
+                    'KmsAlias': dataset.KmsAlias,
+                    'importedS3Bucket': dataset.importedS3Bucket,
+                    'importedGlueDatabase': dataset.importedGlueDatabase,
+                    'importedKmsKey': dataset.importedKmsKey,
+                    'importedAdminRole': dataset.importedAdminRole,
+                }
+        except ResourceUnauthorized:
```

Comment:
I'd let this exception bubble up and appear in the error section of the gql response

Reply:
Why? The end user is authorized to get the object, but some fields are restricted

Reply:
They will get the top level object but an exception for the "restricted" subsection, that way it is more consistent to what the rest of the sub-resolvers are doing

Reply:
I'll work on it, it is true that for consistency it would be better to not retrieve anything

Reply:
I have submitted a new version and tested the following with a user with GET permissions and without GET permissions:
- Dataset view and all tabs
- Table view and tabs
- Folder view
- Worksheets dropdowns